### PR TITLE
fix/V1-177 - fix searching

### PR DESCRIPTION
**Pull request by zhuk-vezunchikz**
- Fix problems with incorrect display of the course selected in the search
- Added closing of search results when choosing a course
- Added clearing of search field when choosing a course

### CODE REVIEW

File: frontend/src/components/Layout/Header/SearchCourses/SearchResult/SearchResult.tsx [hunk 1]
```diff
@@ -5,22 +5,30 @@ import { NO_RESULTS } from 'constants/messages';
 
 import { NoSearchResults, SearchResultWrapper } from './styled';
 import SearchResultItem from './SearchResultItem';
+import { IClientCourseIds } from './types';
 
 interface CoursesFound {
   coursesFound: ICourse[];
-  foundInMyCourses?: string;
+  foundInMyCourses?: IClientCourseIds[];
+  handleSearchClose: () => void;
```
Comment (irina-sinitsyna) [inline frontend/src/components/Layout/Header/SearchCourses/SearchResult/SearchResult.tsx:13]: order

File: frontend/src/components/Layout/Header/SearchCourses/SearchResult/SearchResult.tsx [hunk 2]
```diff
@@ -5,22 +5,30 @@ import { NO_RESULTS } from 'constants/messages';
 
 import { NoSearchResults, SearchResultWrapper } from './styled';
 import SearchResultItem from './SearchResultItem';
+import { IClientCourseIds } from './types';
 
 interface CoursesFound {
```
Comment (irina-sinitsyna) [inline frontend/src/components/Layout/Header/SearchCourses/SearchResult/SearchResult.tsx:10]:
```suggestion
interface ICoursesFound {
```

File: frontend/src/components/Layout/Header/SearchCourses/SearchResult/SearchResult.tsx [hunk 3]
```diff
@@ -5,22 +5,30 @@ import { NO_RESULTS } from 'constants/messages';
 
 import { NoSearchResults, SearchResultWrapper } from './styled';
 import SearchResultItem from './SearchResultItem';
+import { IClientCourseIds } from './types';
```
Comment (irina-sinitsyna) [inline frontend/src/components/Layout/Header/SearchCourses/SearchResult/SearchResult.tsx:8]:
```suggestion
import { IClientCourseId } from './types';
```

File: frontend/src/components/Layout/Header/SearchCourses/SearchResult/SearchResultContainer.tsx [hunk 4]
```diff
@@ -1,36 +1,46 @@
 import React, { useEffect, useState } from 'react';
```
Comment (irina-sinitsyna) [inline frontend/src/components/Layout/Header/SearchCourses/SearchResult/SearchResultContainer.tsx:1]:
```suggestion
import { useEffect, useState } from 'react';
```

File: frontend/src/components/Layout/Header/SearchCourses/SearchResult/SearchResultContainer.tsx [hunk 5]
```diff
@@ -1,36 +1,46 @@
...
 
 import { ICourse } from 'types/course';
-import { ClientCourse } from 'types/clientCourse';
 import useSearchClientCourses from 'api/myCourses/searchClientCourses';
 
 import SearchResult from './SearchResult';
+import { IClientCourseIds } from './types';
 
-interface CoursesFound {
+interface Props {
```
Comment (irina-sinitsyna) [inline frontend/src/components/Layout/Header/SearchCourses/SearchResult/SearchResultContainer.tsx:9]:
```suggestion
interface IProps {
```

Comment (irina-sinitsyna) [reply frontend/src/components/Layout/Header/SearchCourses/SearchResult/SearchResultContainer.tsx:9]: also shouldn't it be moved to ./types ?

File: frontend/src/components/Layout/Header/SearchCourses/SearchResult/SearchResultContainer.tsx [hunk 6]
```diff
@@ -1,36 +1,46 @@
...
-      if (courses[clientCourseIndex].course.title === title) {
-        setFoundInMyCourses(courses[clientCourseIndex]._id);
-        break;
-      }
-    }
-  };
+const SearchResultContainer: React.FC<Props> = ({ coursesFound, handleSearchClose }) => {
+  const [foundInMyCoursesIds, setFoundInMyCoursesIds] = useState<IClientCourseIds[]>([]);
 
   const { data: clientCourses } = useSearchClientCourses();
```
Comment (irina-sinitsyna) [inline frontend/src/components/Layout/Header/SearchCourses/SearchResult/SearchResultContainer.tsx:17]:
```suggestion
  const { data: clientCoursesResponse } = useSearchClientCourses();
```

Comment (irina-sinitsyna) [reply frontend/src/components/Layout/Header/SearchCourses/SearchResult/SearchResultContainer.tsx:17]: or clientCoursesData

File: frontend/src/components/Layout/Header/SearchCourses/SearchResult/SearchResultItem/SearchResultItem.tsx [hunk 7]
```diff
@@ -9,17 +8,22 @@ import { LABEL_MESSAGE } from 'constants/messages';
...
 
-import { Image, CourseTitle, SearchResultCourse } from './styled';
+import { Image, CourseTitle, SearchResultCourse, RedirectButton } from './styled';
 
 interface CourseFound {
   course: ICourse;
   addDivider: boolean;
   status?: string;
-  redirectTo: string;
+  handleSelectFoundCourse: (event: React.MouseEvent<HTMLElement>) => void;
```
Comment (irina-sinitsyna) [inline frontend/src/components/Layout/Header/SearchCourses/SearchResult/SearchResultItem/SearchResultItem.tsx:17]: order

File: frontend/src/components/Layout/Header/SearchCourses/SearchResult/SearchResultItem/SearchResultItemContainer.tsx [hunk 8]
```diff
@@ -1,31 +1,46 @@
...
+import { useNavigate } from 'react-router';
 
 import { ICourse } from 'types/course';
 import transformRoute from 'utils/helpers/paths/transformRoute';
 import { PATHS } from 'constants/routes';
 
 import SearchResultItem from './SearchResultItem';
+import { IClientCourseIds } from '../types';
 
 interface CoursesFound {
```
Comment (irina-sinitsyna) [inline frontend/src/components/Layout/Header/SearchCourses/SearchResult/SearchResultItem/SearchResultItemContainer.tsx:11]:
```suggestion
interface ICoursesFound {
```

File: frontend/src/components/Layout/Header/SearchCourses/SearchResult/SearchResultItem/SearchResultItemContainer.tsx [hunk 9]
```diff
@@ -1,31 +1,46 @@
...
   course: ICourse;
   addDivider: boolean;
-  foundInMyCoursesId?: string;
+  foundInMyCoursesId?: IClientCourseIds[];
+  handleSearchClose: () => void;
```
Comment (irina-sinitsyna) [inline frontend/src/components/Layout/Header/SearchCourses/SearchResult/SearchResultItem/SearchResultItemContainer.tsx:15]: order

**Review by irina-sinitsyna — CHANGES_REQUESTED** [review]

**Review by k-krasilnikova — APPROVED** [review]

**Pull request merged**